src/drive/mobile/modules/settings/components/SelectMediaBuckets.jsx

```diff
@@ -0,0 +1,108 @@
+/* global cozy */
+import React from 'react'
+import { connect } from 'react-redux'
+import Main from 'drive/web/modules/layout/Main'
+import ScrollToTop from 'drive/web/modules/navigation/ScrollToTop'
+import styles from '../styles'
+import Toggle from 'cozy-ui/react/Toggle'
+import { getMediaBuckets, addMediaBucket, delMediaBucket } from '../duck'
+import { getMediaBuckets as getAllMediaBuckets } from 'drive/mobile/lib/media'
+
+const { BarCenter } = cozy.bar
+
+class SelectMediaBuckets extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      mediaBuckets: []
+    }
+  }
+
+  createHandle(item) {
+    const self = this
+    return checked => {
+      if (checked) {
+        self.props.addMediaBucket(item)
+      } else {
+        self.props.delMediaBucket(item)
+      }
+    }
+  }
+
+  handleChange(e) {
+    const item = e.target.id
+    const isChecked = e.target.checked
+    if (isChecked) {
+      this.setState(state => ({
+        selectedBuckets: state.selectedBuckets.add(item)
+      }))
+    } else {
+      this.setState(state => {
+        const newSelectedBuckets = new Set(state.selectedBuckets)
+        newSelectedBuckets.delete(item)
+        return {
+          selectedBuckets: newSelectedBuckets
+        }
+      })
+    }
+  }
+
+  componentDidMount() {
+    var self = this
+    getAllMediaBuckets().then(data => {
+      self.setState({
+        mediaBuckets: data
+      })
+    })
+  }
+
+  render() {
+    return (
+      <Main>
+        <ScrollToTop />
+        <BarCenter>
+          <h2 className={styles['settings__title']}>Select Media Buckets</h2>
+        </BarCenter>
+        {this.state.mediaBuckets.map(item => (
+          <div className={styles['settings__subcategory']} key={item}>
+            <label
+              htmlFor={item}
+              className={styles['settings__subcategory__label']}
+            >
+              {item}
+            </label>
+            <label
+              htmlFor={item}
+              className={styles['settings__subcategory__item']}
+            >
+              <Toggle
+                id={item}
+                checked={this.props.selectedBuckets.has(item)}
+                onToggle={this.createHandle(item)}
```

Comment:
OK now I understand. You can `bind` this.createHandle to `this` in order to have the right context when you call it. 

```jsx 
constructor(){
this.createHandle = this.createHandle.bind(this, item) 
}
```

Or : 
```jsx 
onToggle={ () => this.createHandle(item) }
```
since arrow function auto bind to this

Reply:
I used an arrow function with a `checked` argument :
```
onToggle={checked => this.createHandle(item, checked)}
```